feature/quicksettings/src/main/java/com/google/jetpackcamera/feature/quicksettings/ui/QuickSettingsComponents.kt

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jetpackcamera.feature.quicksettings.ui
+
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.material3.Icon
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.google.jetpackcamera.feature.quicksettings.CameraAspectRatio
+import com.google.jetpackcamera.feature.quicksettings.CameraFlashMode
+import com.google.jetpackcamera.feature.quicksettings.CameraLensFace
+import com.google.jetpackcamera.quicksettings.R
+import com.google.jetpackcamera.settings.model.FlashModeStatus
+import kotlin.math.min
+
+
+// completed components
+
+@Composable
+fun ExpandedQuickSetRatio(
+    setRatio: (/*TODO: insert ratio function*/) -> Unit,
+    currentRatio: Int
+) {
+    val buttons: Array<@Composable () -> Unit> = arrayOf(
+        { QuickSetRatio(onClick = { setRatio() }, ratio = 1, currentRatio = currentRatio) },
+        { QuickSetRatio(onClick = { setRatio() }, ratio = 2, currentRatio = currentRatio) },
+        { QuickSetRatio(onClick = { setRatio() }, ratio = 3, currentRatio = currentRatio) }
+
+    )
+    ExpandedQuickSetting(quickSettingButtons = buttons)
+}
+
+@Composable
+fun QuickSetRatio(onClick: () -> Unit, ratio: Int, currentRatio: Int) {
```

Comment:
we should discuss about QuickSettingUiModel vs. QuickSet***Components. The reason I did it that way is so that we wouldn't have to duplicate code and we can keep track of all the enums and their relationship to the component in a single file. But I see merit doing this way too

Reply:
yeah like we had spoken about earlier, I think this approach would simplify the process of creating/removing quick settings, and improve readability for anyone new looking at the code